### ISSUE TITLE
Issues with USB communication on AMD AM4 Mainboard (X370 + Ryzen)

### ISSUE THREAD

**Issue by hpingel** [davidgiven/fluxengine#98]
Hi David,

Now this is the kind of issue nobody wants to read on his own project's bugtracker because there's not much more to be done than reading, shrugging and sighing. ;-)

With best intentions I tried to run FluxEngine on my AMD Ryzen box (ASUS Prime X370-PRO, Ryzen 1700, Nvidia GTX 1060). I had issues whenever USB bulk transfer is required. I tried different USB cables and different USB ports.

- `fluxengine` command displays help normal.
- `fluxengine rpm` command runs normal.
- `fluxengine read ibm` results either in "USB underrun" or "Error: bad USB reply"
- `fluxengine testbulktransport` returns as last line: "Error: data transfer corrupted"

Error messages are copied from Linux on the box, but beware:

On this box I have Ubuntu 18.04 amd64 **and** I also have Windows 10 64bit. So I was able to test the FluxEngine on both OSses. That's why I think the issue is related to my hardware or my hardware configurations (have to check UEFI settings).

The same FluxEngine works fine on my Lenovo Notebook with Windows 10. On The notebook I get 750 KB/s according to `fluxengine testbulktransport`.

On the problematic X370 box I only seem to get 600-670KB/s.

I'm not asking for a solution, I just wanted to document that I have a problem here. Maybe in a week's time I will stumble across the solution - then I will document it here.

On Linux the FluxEngine is compiled locally, on Windows I used the latest experimental binary.

Dmesg on Linux shows me:
```
[    1.420892] usb 3-2: new full-speed USB device number 2 using xhci_hcd
[    1.839696] usb 3-2: New USB device found, idVendor=1209, idProduct=6e00, bcdDevice= 0.01
[    1.839698] usb 3-2: New USB device strings: Mfr=2, Product=1, SerialNumber=128
[    1.839699] usb 3-2: Product: FluxEngine
[    1.839700] usb 3-2: Manufacturer: Cowlark Technologies
[    1.839701] usb 3-2: SerialNumber: xxxxxxxxxxxxxxxxxxxxxx
```

On Linux, I also tried to update libusb-1.0.0 packages to more recent versions. Syslog and dmesg don't show any obvious thing. All I can see while I'm issueing FluxEngine  commands is:

```
Aug 22 19:19:37 ryzentux upowerd[1386]: unhandled action 'bind' on /sys/devices/pci0000:00/0000:00:01.3/0000:02:00.2/0000:03:04.0/0000:07:00.0/usb3/3-2/3-2:1.0
Aug 22 19:19:39 ryzentux upowerd[1386]: unhandled action 'unbind' on /sys/devices/pci0000:00/0000:00:01.3/0000:02:00.2/0000:03:04.0/0000:07:00.0/usb3/3-2/3-2:1.0
```
But the issue also occurs on Windows 10.

My current ideas are:
- It could be related to USB power management of my system or maybe related to several different USB chips I have on my mainboard.
- Electric problem due to floppy being powered by PSU of the box and also connected to box via USB.

**Comment by davidgiven** [comment]
At least some of this is a duplicate of #100, but that's really a symptom, not a cause.

`testbulktransfer` reporting corrupted data is extremely problematic, as USB transfers should be reliable. The bandwidth is low, too. My gut says that there could be something wrong with the USB setup, but my gut also says that _The Flash_ is worth watching, so...

I see that you've tried different cables. Have you tried a powered hub? Is the FluxEngine itself running from the FDD power supply? Do you still get problems with `testbulktransfer` if the FluxEngine is _not_ powered from the FDD power supply (and with the FDD disconnected)?

**Comment by hpingel** [comment]
Just dumping some brainstorming stuff related to my AMD Ryzen 1700 + USB kernel driver issues. As a first step will just try to turn IOMMU off to check if it improves the situation at all.

Getting:
`xhci_hcd 0000:07:00.0: WARN Set TR Deq Ptr cmd failed due to incorrect slot or ep state.`

Leads me to:
- https://bugzilla.kernel.org/show_bug.cgi?id=202541
- https://github.com/ZerBea/hcxdumptool/issues/42
- https://bugzilla.kernel.org/show_bug.cgi?id=202241

**Comment by hpingel** [comment]
Good news:

- Problem on Linux solved by putting a cheap USB 2.0 Hub in-between FluxEngine and the PC. Some would call it a workaround. I just sigh.
- Problem on Windows 10 probably solved by installing AMD Chipset Drivers which will replace the standard Windows 10 USB drivers. If it works I will not reopen this ticket.
